drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -310,15 +310,14 @@ static inline uint8_t nrfx_addr_to_ep(nrfx_usbd_ep_t ep)
 
 static inline bool ep_is_valid(const u8_t ep)
 {
-	u8_t ep_num = NRF_USBD_EP_NR_GET(ep);
```

Comment:
Instead of removing it, you should change it to:
```
u8_t ep_num = ep & ~USB_EP_DIR_MASK;
```
Then you won't need these next two changes in this function.

But actually, `NRF_USBD_EP_NR_GET` is also used inside `NRF_USBD_EPISO_CHECK`, so this function still won't properly detect that for instance 0x18 is not a valid endpoint number.
I think that both occurrences of `NRF_USBD_EPISO_CHECK(ep)` below should be replaced with `ep_num == NRF_USBD_EPISO_FIRST` in order to solve this.

Reply:
indeed, have overlooked it